Microsoft.Toolkit.Uwp.UI/Extensions/TextBoxRegEx/TextBoxRegex.cs

```diff
@@ -109,10 +117,21 @@ private static void ValidateTextBox(TextBox textbox, bool force = true)
             {
                 if (force)
                 {
-                    var validationModel = (ValidationMode)textbox.GetValue(ValidationModeProperty);
-                    if (validationModel == ValidationMode.Forced)
+                    if (textbox.Text != string.Empty)
                     {
-                        textbox.Text = string.Empty;
+                        var validationModel = (ValidationMode)textbox.GetValue(ValidationModeProperty);
+                        if (validationModel == ValidationMode.Forced || validationModel == ValidationMode.Instantly)
+                        {
+                            textbox.Text = string.Empty;
+                        }
+                        else if (validationModel == ValidationMode.InstantlyNumber)
```

Comment:
Would BeforeTextChanging be beneficial for this? That way you don't have to change the text.

Reply:
Yes, I will look into BeforeTextChanging. The question is, the event needs the newest SDK? What is the best practise for handle that?
https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.textboxbeforetextchangingeventargs

Reply:
Could check if the contract is present and act accordingly. I think we could do that in a separate PR though. @nmetulev